x/nft/keeper/denom.go

```diff
@@ -64,3 +64,21 @@ func (k Keeper) GetDenoms(ctx sdk.Context) (denoms []types.Denom) {
 	}
 	return denoms
 }
+
+// IsDenomCreator checks if address is the creator of Denom
+// Return the Denom if true, an error otherwise
+// nolint: interfacer
+func (k Keeper) IsDenomCreator(ctx sdk.Context, denomID string, address sdk.AccAddress) (types.Denom, error) {
+	denom, err := k.GetDenom(ctx, denomID)
+	if err != nil {
+		return types.Denom{}, err
+	}
+
+	addressStr := address.String()
+
+	if addressStr != denom.Creator {
```

Comment:
I'd probably add a comment here why the string comparison is ok -- is the denom.creator guaranteed/checked to be a valid address or is it possible one can set it to invalid string (empty, some random...)?

Reply:
Changed to comparing concrete address types